### DISCUSSION TITLE
求问如何主动向群聊/用户发送消息

### DISCUSSION THREAD

**Discussion by LightningMaker** [Hoshikawa-Kaguya/Sora#29] [Q&A] [answered]
示例中只有回复消息的。如果主动发送消息/定时发送消息的话，有没有什么简便的方法
谢谢

**Comment by Yukari316** [comment, marked as answer]
可以尝试将 `SoraApi` 这个实例保存下来然后在之后的定时业务中调用实例中的API（比如塞进容器里之类的）
比如 用 `SendGroupMessage` 来发送一些要主动发送的消息
框架在处理几乎所有的上报事件的时候都会创建一个 `SoraApi` 实例来给上层API使用
 `SoraApi` 也提供了几乎所有的底层API方法
需要注意的是 `SoraApi` 是对单一连接单一服务所创建的实例，所以在 `SoraApi` 所对应的服务连接失效后，实例中的API也均会失效，这一个需要注意

不过后续版本应该会提供一个动态创建 `SoraApi` 实例的方法以供定时任务需要，不过这个东西还在后续的架构考虑之中应该不会太快去支持

**Comment by AnnualWater** [comment]
这种需求可以考虑去调用go-cqhttp的Web API来解决，在go-cqhttp里配置好HTTP的配置，在你的服务里使用POST请求来主动发送信息。
不过这得需要你去看go-cqhttp或者说onebot的文档
如果有token的话，记得在请求的时候不要忘了它

**Reply by Yukari316** [reply]
确实（
